superbench/benchmarks/model_benchmarks/pytorch_base.py

```diff
@@ -35,10 +44,243 @@ def __init__(self, name, parameters=''):
+        """Centralized logic for recording per-step loss and periodic fingerprints for deterministic runs.
+
+        Args:
+            curr_step (int): Current training step.
+            loss (torch.Tensor or float): Loss value for this step.
+            logits (torch.Tensor or float): Logits output for this step (sample 0).
+            periodic (dict): Dictionary to store periodic fingerprints ('loss', 'act_mean', 'step').
+            check_frequency (int): Frequency for fingerprint logging.
+        """
+        # Record per-step loss for determinism checks (for full history)
+        try:
+            v = float(loss.detach().item()) if hasattr(loss, 'detach') else float(loss)
+        except Exception:
+            v = None
+        # Periodic fingerprint logging
+        if getattr(self._args, 'deterministic', False) and (curr_step % check_frequency == 0):
+            # 1) Loss fingerprint (only at fingerprinting frequency)
+            try:
+                if 'loss' in periodic and v is not None:
+                    periodic['loss'].append(v)
+                logger.info(f'Loss at step {curr_step}: {v}')
+                periodic['step'].append(curr_step)
+            except Exception:
+                pass
+            # 2) Tiny activation fingerprint: mean over logits for sample 0
+            try:
+                if logits is not None:
+                    act_mean = (
+                        float(logits[0].detach().float().mean().item())
+                        if hasattr(logits[0], 'detach') else float(logits[0])
+                    )
+                    logger.info(f'ActMean at step {curr_step}: {act_mean}')
+                    periodic['act_mean'].append(act_mean)
+            except Exception:
+                pass
+
+    def _finalize_periodic_logging(self, duration, periodic, info_key='loss'):
+        """Finalize periodic logging and return results tuple for training step."""
+        info = {info_key: periodic.get(info_key, [])}
+        self._model_run_losses = list(periodic.get(info_key, []))
+        self._model_run_periodic = dict(periodic)
+        return (duration, info)
+
+    def _benchmark(self):
+        """Run the benchmark then handle post-run model log save/compare."""
+        ok = super()._benchmark()
+        self._post_run_model_log()
+        return ok
+
+    def add_parser_arguments(self):
+        """Add PyTorch model benchmark-specific arguments to the argument parser."""
+        super().add_parser_arguments()
+        self._parser.add_argument(
+            '--generate-log',
+            '--generate_log',
+            dest='generate_log',
+            action='store_true',
+            default=False,
+            help='Save fingerprint log to file.',
+        )
+        self._parser.add_argument(
+            '--log-path',
+            '--log_path',
+            dest='log_path',
+            type=str,
+            default=None,
+            help='Path to save or load fingerprint log.',
```

Comment:
Looks this argument is not used when `load` log in your example:
```
--model <model_from_MODEL_CHOICES> --generate-log --log-path ./outputs/determinism_ref.json
--model <model_from_MODEL_CHOICES> --compare-log ./outputs/determinism_ref.json
```

Should we unify the usage? Like
```
--model <model_from_MODEL_CHOICES> --compare-log --log-path ./outputs/determinism_ref.json
```